Added section on maintainer-pain

JesperIRL opened the pull request:
Added section on maintainer-pain

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * [Igor Ignatyev](https://openjdk.java.net/census#iignatyev) (@iignatev - no project role) ⚠️ Review applies to ac1986b466465c6798e6b6590930728300240167
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - no project role) ⚠️ Review applies to ac1986b466465c6798e6b6590930728300240167
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - no project role) ⚠️ Review applies to dc0909b4fd7ceb1b53766202d771da76dcc15629
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to dc0909b4fd7ceb1b53766202d771da76dcc15629


### Download
`$ git fetch https://git.openjdk.java.net/guide pull/37/head:pull/37`
`$ git checkout pull/37`

iignatev commented on src/index.md:
```diff
@@ -969,6 +969,24 @@ reverse effect of earlier changeset
+
+If you have a [`maintainer-pain`](#maintainer-pain) bug assigned to you please consider fixing it asap. If you chose not to work on the issue, you should at least be aware that you are choosing to waste others' engineering time and people will be affected by this choice.
+
+As with any issue the best way to deal with a [`maintainer-pain`](#maintainer-pain) issue is to fix it. Another way to reduce the noise is to ProblemList the failing test. This is a viable option if there is a limited set of tests that are failing and the bug is actively investigated. ProblemListing a test is a regression in our test coverage so this should not be taken lightly, but it works well ...
```
should we refer to `ProblemListing` section here?

JesperIRL replied:
Yes, that is a good idea!

iignatev reviewed (approved):
@JesperIRL , looks good to me in general. however, I'm not sure that every occurrence of `maintainer-pain` should be a reference, usually, it's enough to have just 1st appearance as a reference.

-- Igor

JesperIRL commented:
Just for the record, the maintainer-pain label and the use of it was discussed on jdk-dev here: https://mail.openjdk.java.net/pipermail/jdk-dev/2018-September/001929.html

JesperIRL commented:
> @JesperIRL , looks good to me in general. however, I'm not sure that every occurrence of `maintainer-pain` should be a reference, usually, it's enough to have just 1st appearance as a reference.

Currently all JBS labels in the Guide are linked to a description of the label. I see your point, it's unnecessary to link every single occurrence in such a short section, but if you don't mind terribly I'd like to maintain the invariant that every JBS label mentioned is linked ot the description.

Do you find it extremely ugly with all the links?

iignatev commented:
> > @JesperIRL , looks good to me in general. however, I'm not sure that every occurrence of `maintainer-pain` should be a reference, usually, it's enough to have just 1st appearance as a reference.
> 
> Currently all JBS labels in the Guide are linked to a description of the label. I see your point, it's unnecessary to link every single occurrence in such a short section, but if you don't mind terribly I'd like to maintain the invariant that every JBS label mentioned is linked ot the description.
> 
> Do you find it extremely ugly with all the links?

I haven't seen it rendered, so for the present, I don't find it ugly at all ;)

JesperIRL commented:
These comments came to me through other channels. Saving here for tracking purposes.

> Maybe a better heading too as “Handling Maintainer Pain” will be alien to most people.

Renamed it to "Tracking noisy issues (maintainer-pain)" 

> I think the phrase “engineering time” should be dropped as I don’t think it make sense in the context of OpenJDK.

Right. Fixed.

> Also I thought we decided early on to talk about excluding tests, not “ProblemListing”.

I hear different opinions on this. I have removed all instances of ProblemListing in this section, but the section about excluding tests is still using this term.

> It might be useful to outline a few board examples of issues to explain what this section is about. I think it’s about frequent failures, maybe on a specific platform, maybe specific to one vendor’s test infrastructure. It might is some underlying bug that causes several tests to fail intermittently. Maybe it’s a change causes a test failure in a faraway place and the failure isn’t acted on quickly.  So more text to explain what it is about would be useful.

I broke the text into separate examples and added the cases you brought up as well.

> I think the main point you want to get across is you want to drop test failures that would waste time investigating because it’s already tracked.

That's one use case but there are other. For instance a bug that causes test failures all over the place - even if everyone know what but it is and no time is spent investigating the issue the failures are clearly a pain to everyone.

prrace reviewed (changes requested):
I really don't think this entire section should be there. maintainer-pain is something that was discussed, and it made no sense, so it was discussed again and still seemed marginal and maybe was interesting to hotspot but has no broad acceptance or adoption.

JesperIRL commented:
> I really don't think this entire section should be there. maintainer-pain is something that was discussed, and it made no sense, so it was discussed again and still seemed marginal and maybe was interesting to hotspot but has no broad acceptance or adoption.

Unless there is another mail thread about this that I've forgotten (or not seen) than the one I linked to above, I don't agree that the idea of having a label to indicate that an issue is urgent was deemed to make no sense. There was a misunderstanding in the early part of the thread due to initial bad naming of the label, but once people understood what the label was about I don't see anyone objecting to the use of such a label. There are several positive comments in that thread suggesting that such a label would make sense.

The label is only informational, there is no process that mandate that the label is used, but if someone set that label on your issue, you know what it means. It's up to you to decide wether to care or not, but as the text here says: "If you chose not to work on the issue, you should at least be aware that you are choosing to waste others' time and people will be affected by this choice."

As for adoption you are absolutely right. The label hasn't been used a lot and I guess that could either be a good sign - we don't have many such bugs - or a sign of no one knowing about the label.

After some mulling I do think I'd vote to include this section. I'd be willing to re-think the name of the label, "urgent" was suggested in the thread and in hind-sight I think it's a better label in some sense (more clear).

JesperIRL commented on src/index.md:
```diff
@@ -969,6 +969,34 @@ reverse effect of earlier changeset
+The JBS label [`maintainer-pain`](#maintainer-pain) is used to tag bugs that for some reason is wasting time or in other ways are causing pain for the OpenJDK maintainers. Examples of issues that could be considered a pain:
+
+* A bug that occurs frequently in testing, maybe on a specific platform, maybe specific to one vendor's test infrastructure, and requires that many maintainers investigate the failure in different test runs just to realize it's the same issue as have been seen and reported before, or worse, don't realize it's a known issue and file a duplicate bug in JBS.
+
```
Fixed.

JesperIRL commented:
I moved the entire description to the JBS Label Dictionary instead. This makes it more clear that this is not really a process, but rather an explanation of how to interpret and use the label. I also removed some wording that made the text sound as a process.

@prrace Does this change make you more comfortable with the text?

JesperIRL commented:
/integrate
